crates/forge/src/fuzzer.rs

```diff
@@ -0,0 +1,171 @@
+use cairo_felt::Felt252;
+use num_bigint::{BigUint, RandBigInt};
+use num_traits::One;
+use rand::rngs::StdRng;
+use rand::{Rng, SeedableRng};
+use std::default::Default;
+
+#[derive(Default)]
+#[allow(clippy::module_name_repetitions)]
+pub struct FuzzerRunParams {
+    /// Inclusive value
+    low: BigUint,
+    /// Exclusive value
+    high: BigUint,
+    /// Number of arguments
+    args_number: usize,
+    /// Total number of runs
+    runs_number: u32,
+    /// Number of already executed runs
+    runs_executed: u32,
+    /// Runs in which arguments have min value e.g. `run_with_min_value[0] = 5`
+    /// means that the first argument has the lowest possible value in 5th run
+    runs_with_min_value: Vec<u32>,
+    /// Runs in which arguments have max value e.g. `run_with_min_value[0] = 5`
+    /// means that the first argument has the highest possible value in 5th run
+    runs_with_max_value: Vec<u32>,
+}
+
+#[allow(clippy::module_name_repetitions)]
+pub struct RandomFuzzer {
+    rng: StdRng,
+    fuzzer_run_params: FuzzerRunParams,
+}
+
+impl RandomFuzzer {
+    pub fn new(
+        seed: u64,
+        runs_number: u32,
+        args_number: usize,
+        low: BigUint,
+        high: BigUint,
+    ) -> Self {
+        let rng = StdRng::seed_from_u64(seed);
+
+        let mut fuzzer = RandomFuzzer {
+            rng,
+            fuzzer_run_params: FuzzerRunParams::default(),
+        };
+        fuzzer.set_fuzzer_run_params(runs_number, args_number, low, high);
+
+        fuzzer
+    }
+
+    fn set_fuzzer_run_params(
+        &mut self,
+        runs_number: u32,
+        args_number: usize,
+        low: BigUint,
+        high: BigUint,
+    ) {
+        assert!(low < high);
+        assert!(runs_number >= 3);
+
+        self.fuzzer_run_params.runs_executed = 0;
+        self.fuzzer_run_params.low = low;
+        self.fuzzer_run_params.high = high;
+        self.fuzzer_run_params.runs_number = runs_number;
+        self.fuzzer_run_params.args_number = args_number;
+
+        let runs_with_min_value: Vec<u32> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .map(|_| self.rng.gen_range(1..=self.fuzzer_run_params.runs_number))
+            .collect();
+        let runs_with_max_value: Vec<u32> = vec![0; self.fuzzer_run_params.args_number]
+            .into_iter()
+            .zip(&runs_with_min_value)
+            .map(|(_, &run_with_min)| {
+                let run_with_max = self.rng.gen_range(1..=self.fuzzer_run_params.runs_number);
+                if run_with_max == run_with_min {
+                    run_with_min + 1 % runs_number
+                } else {
+                    run_with_max
+                }
+            })
+            .collect();
```

Comment:
Wouldn't it be easier to just throw in `[max, min, rand1, rand2..., rand(n-2)]` ?

Reply:
Actually true, will change it

Reply:
The idea was to be able to test it in different combinations to create stronger test cases, this way you will only test 0 with other 0s.

Reply:
Not really, we want to have max and min values in random runs. The proposed solution is fine for `args_number == 1`, otherwise we will have first run will all args set to max, then 2nd one with all args set to min.

TL;DR: for 3 args, 3 runs, `low=0`, `high=10` currently we can have sth like:
```
[10, 0, 1]
[0, 2, 0]
[5, 10, 10]
```
Using the proposed solution we will have
```
[10, 10,  10]
[0, 0, 0]
[5, 3, 2]
```
so max and min values are always tested together :/